Build failing  on Kernel 4.14 on musl libc

See [build log](https://github.com/netdata/kernel-collector/pull/22/checks?check_run_id=431468725)

For the lazy:

```
2020-02-07T09:09:04.8425998Z cd ../library/ && make all;
2020-02-07T09:09:04.8449883Z make[1]: Entering directory '/kernel-collector/library'
2020-02-07T09:09:04.8584153Z cd ../libbpf_0_0_1/src/ && make all; \
2020-02-07T09:09:04.8585119Z mkdir -p root; DESTDIR=root make install_headers;
2020-02-07T09:09:04.8619124Z make[2]: Entering directory '/kernel-collector/libbpf_0_0_1/src'
2020-02-07T09:09:04.9016704Z Package libelf was not found in the pkg-config search path.
2020-02-07T09:09:04.9017052Z Perhaps you should add the directory containing `libelf.pc'
2020-02-07T09:09:04.9017172Z to the PKG_CONFIG_PATH environment variable
2020-02-07T09:09:04.9017551Z Package 'libelf', required by 'virtual:world', not found
2020-02-07T09:09:04.9036551Z cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -fno-stack-protector -I /usr/src/linux/usr/include   -c bpf.c -o bpf.o
2020-02-07T09:09:04.9855953Z cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -fno-stack-protector -I /usr/src/linux/usr/include   -c btf.c -o btf.o
2020-02-07T09:09:05.1773957Z cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -fno-stack-protector -I /usr/src/linux/usr/include   -c libbpf.c -o libbpf.o
2020-02-07T09:09:05.2001601Z In file included from /usr/src/linux/usr/include/linux/byteorder/little_endian.h:13,
2020-02-07T09:09:05.2001864Z                  from /usr/src/linux/usr/include/asm/byteorder.h:5,
2020-02-07T09:09:05.2001992Z                  from /usr/src/linux/usr/include/linux/perf_event.h:20,
2020-02-07T09:09:05.2002112Z                  from libbpf.c:33:
2020-02-07T09:09:05.2003921Z /usr/src/linux/usr/include/linux/swab.h:161:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2004097Z   161 | static __always_inline __u16 __swab16p(const __u16 *p)
2020-02-07T09:09:05.2004199Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2004600Z /usr/src/linux/usr/include/linux/swab.h:161:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__swab16p'
2020-02-07T09:09:05.2004730Z   161 | static __always_inline __u16 __swab16p(const __u16 *p)
2020-02-07T09:09:05.2004853Z       |                              ^~~~~~~~~
2020-02-07T09:09:05.2005177Z /usr/src/linux/usr/include/linux/swab.h:174:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2005303Z   174 | static __always_inline __u32 __swab32p(const __u32 *p)
2020-02-07T09:09:05.2005414Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2005786Z /usr/src/linux/usr/include/linux/swab.h:174:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__swab32p'
2020-02-07T09:09:05.2005907Z   174 | static __always_inline __u32 __swab32p(const __u32 *p)
2020-02-07T09:09:05.2006023Z       |                              ^~~~~~~~~
2020-02-07T09:09:05.2006360Z /usr/src/linux/usr/include/linux/swab.h:187:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2006488Z   187 | static __always_inline __u64 __swab64p(const __u64 *p)
2020-02-07T09:09:05.2006587Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2006956Z /usr/src/linux/usr/include/linux/swab.h:187:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__swab64p'
2020-02-07T09:09:05.2007295Z   187 | static __always_inline __u64 __swab64p(const __u64 *p)
2020-02-07T09:09:05.2007405Z       |                              ^~~~~~~~~
2020-02-07T09:09:05.2012488Z /usr/src/linux/usr/include/linux/swab.h: In function '__swab16s':
2020-02-07T09:09:05.2015142Z /usr/src/linux/usr/include/linux/swab.h:235:7: warning: implicit declaration of function '__swab16p'; did you mean '__swab16s'? [-Wimplicit-function-declaration]
2020-02-07T09:09:05.2015320Z   235 |  *p = __swab16p(p);
2020-02-07T09:09:05.2015628Z       |       ^~~~~~~~~
2020-02-07T09:09:05.2015739Z       |       __swab16s
2020-02-07T09:09:05.2015829Z /usr/src/linux/usr/include/linux/swab.h: At top level:
2020-02-07T09:09:05.2018759Z /usr/src/linux/usr/include/linux/swab.h:242:23: error: expected ';' before 'void'
2020-02-07T09:09:05.2019114Z   242 | static __always_inline void __swab32s(__u32 *p)
2020-02-07T09:09:05.2019378Z       |                       ^~~~~
2020-02-07T09:09:05.2019590Z       |                       ;
2020-02-07T09:09:05.2026711Z /usr/src/linux/usr/include/linux/swab.h: In function '__swab32s':
2020-02-07T09:09:05.2027401Z /usr/src/linux/usr/include/linux/swab.h:247:7: warning: implicit declaration of function '__swab32p'; did you mean '__swab32s'? [-Wimplicit-function-declaration]
2020-02-07T09:09:05.2027682Z   247 |  *p = __swab32p(p);
2020-02-07T09:09:05.2027914Z       |       ^~~~~~~~~
2020-02-07T09:09:05.2028122Z       |       __swab32s
2020-02-07T09:09:05.2032119Z /usr/src/linux/usr/include/linux/swab.h: At top level:
2020-02-07T09:09:05.2032863Z /usr/src/linux/usr/include/linux/swab.h:255:23: error: expected ';' before 'void'
2020-02-07T09:09:05.2033133Z   255 | static __always_inline void __swab64s(__u64 *p)
2020-02-07T09:09:05.2033352Z       |                       ^~~~~
2020-02-07T09:09:05.2033552Z       |                       ;
2020-02-07T09:09:05.2037149Z /usr/src/linux/usr/include/linux/swab.h: In function '__swab64s':
2020-02-07T09:09:05.2038266Z /usr/src/linux/usr/include/linux/swab.h:260:7: warning: implicit declaration of function '__swab64p'; did you mean '__swab64s'? [-Wimplicit-function-declaration]
2020-02-07T09:09:05.2038528Z   260 |  *p = __swab64p(p);
2020-02-07T09:09:05.2038747Z       |       ^~~~~~~~~
2020-02-07T09:09:05.2038937Z       |       __swab64s
2020-02-07T09:09:05.2042671Z In file included from /usr/src/linux/usr/include/asm/byteorder.h:5,
2020-02-07T09:09:05.2042991Z                  from /usr/src/linux/usr/include/linux/perf_event.h:20,
2020-02-07T09:09:05.2043229Z                  from libbpf.c:33:
2020-02-07T09:09:05.2043438Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h: At top level:
2020-02-07T09:09:05.2044061Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:44:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2044297Z    44 | static __always_inline __le64 __cpu_to_le64p(const __u64 *p)
2020-02-07T09:09:05.2044522Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2045062Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:44:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_le64p'
2020-02-07T09:09:05.2045330Z    44 | static __always_inline __le64 __cpu_to_le64p(const __u64 *p)
2020-02-07T09:09:05.2045555Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2046017Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:48:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2046298Z    48 | static __always_inline __u64 __le64_to_cpup(const __le64 *p)
2020-02-07T09:09:05.2046522Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2047024Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:48:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__le64_to_cpup'
2020-02-07T09:09:05.2047275Z    48 | static __always_inline __u64 __le64_to_cpup(const __le64 *p)
2020-02-07T09:09:05.2047504Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2048135Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:52:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2048399Z    52 | static __always_inline __le32 __cpu_to_le32p(const __u32 *p)
2020-02-07T09:09:05.2048621Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2049125Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:52:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_le32p'
2020-02-07T09:09:05.2049372Z    52 | static __always_inline __le32 __cpu_to_le32p(const __u32 *p)
2020-02-07T09:09:05.2049676Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2050158Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:56:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2050419Z    56 | static __always_inline __u32 __le32_to_cpup(const __le32 *p)
2020-02-07T09:09:05.2050621Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2051134Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:56:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__le32_to_cpup'
2020-02-07T09:09:05.2051388Z    56 | static __always_inline __u32 __le32_to_cpup(const __le32 *p)
2020-02-07T09:09:05.2051614Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2052088Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:60:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2052870Z    60 | static __always_inline __le16 __cpu_to_le16p(const __u16 *p)
2020-02-07T09:09:05.2054039Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2054774Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:60:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_le16p'
2020-02-07T09:09:05.2055020Z    60 | static __always_inline __le16 __cpu_to_le16p(const __u16 *p)
2020-02-07T09:09:05.2055248Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2055746Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:64:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2055980Z    64 | static __always_inline __u16 __le16_to_cpup(const __le16 *p)
2020-02-07T09:09:05.2056199Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2056715Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:64:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__le16_to_cpup'
2020-02-07T09:09:05.2056967Z    64 | static __always_inline __u16 __le16_to_cpup(const __le16 *p)
2020-02-07T09:09:05.2057172Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2057652Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:68:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2057900Z    68 | static __always_inline __be64 __cpu_to_be64p(const __u64 *p)
2020-02-07T09:09:05.2058117Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2058645Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:68:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_be64p'
2020-02-07T09:09:05.2058900Z    68 | static __always_inline __be64 __cpu_to_be64p(const __u64 *p)
2020-02-07T09:09:05.2059121Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2059598Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:72:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2059828Z    72 | static __always_inline __u64 __be64_to_cpup(const __be64 *p)
2020-02-07T09:09:05.2060044Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2060539Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:72:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__be64_to_cpup'
2020-02-07T09:09:05.2060799Z    72 | static __always_inline __u64 __be64_to_cpup(const __be64 *p)
2020-02-07T09:09:05.2061002Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2061550Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:76:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2061795Z    76 | static __always_inline __be32 __cpu_to_be32p(const __u32 *p)
2020-02-07T09:09:05.2062141Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2062653Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:76:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_be32p'
2020-02-07T09:09:05.2062901Z    76 | static __always_inline __be32 __cpu_to_be32p(const __u32 *p)
2020-02-07T09:09:05.2063120Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2063580Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:80:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2063903Z    80 | static __always_inline __u32 __be32_to_cpup(const __be32 *p)
2020-02-07T09:09:05.2064130Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2064635Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:80:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__be32_to_cpup'
2020-02-07T09:09:05.2064883Z    80 | static __always_inline __u32 __be32_to_cpup(const __be32 *p)
2020-02-07T09:09:05.2065111Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2065589Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:84:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2065817Z    84 | static __always_inline __be16 __cpu_to_be16p(const __u16 *p)
2020-02-07T09:09:05.2066034Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2066560Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:84:31: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__cpu_to_be16p'
2020-02-07T09:09:05.2066797Z    84 | static __always_inline __be16 __cpu_to_be16p(const __u16 *p)
2020-02-07T09:09:05.2067014Z       |                               ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2067485Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:88:8: error: unknown type name '__always_inline'
2020-02-07T09:09:05.2067726Z    88 | static __always_inline __u16 __be16_to_cpup(const __be16 *p)
2020-02-07T09:09:05.2067928Z       |        ^~~~~~~~~~~~~~~
2020-02-07T09:09:05.2068440Z /usr/src/linux/usr/include/linux/byteorder/little_endian.h:88:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__be16_to_cpup'
2020-02-07T09:09:05.2068707Z    88 | static __always_inline __u16 __be16_to_cpup(const __be16 *p)
2020-02-07T09:09:05.2068925Z       |                              ^~~~~~~~~~~~~~
2020-02-07T09:09:05.2570292Z make[2]: *** [Makefile:91: libbpf.o] Error 1
2020-02-07T09:09:05.2571387Z make[2]: Leaving directory '/kernel-collector/libbpf_0_0_1/src'
2020-02-07T09:09:05.2598672Z make[2]: Entering directory '/kernel-collector/libbpf_0_0_1/src'
2020-02-07T09:09:05.2973301Z Package libelf was not found in the pkg-config search path.
2020-02-07T09:09:05.2973693Z Perhaps you should add the directory containing `libelf.pc'
2020-02-07T09:09:05.2973864Z to the PKG_CONFIG_PATH environment variable
2020-02-07T09:09:05.2974154Z Package 'libelf', required by 'virtual:world', not found
2020-02-07T09:09:05.2991098Z if [ ! -d 'root/usr/include/bpf' ]; then install -d -m 755 'root/usr/include/bpf'; fi; install bpf.h libbpf.h btf.h xsk.h libbpf_util.h -m 644 'root/usr/include/bpf'
2020-02-07T09:09:05.3008469Z make[2]: Leaving directory '/kernel-collector/libbpf_0_0_1/src'
2020-02-07T09:09:05.3010706Z cp ../libbpf_0_0_1/src//libbpf.so.0.0.3 libbpf.so
2020-02-07T09:09:05.3036231Z cp: cannot stat '../libbpf_0_0_1/src//libbpf.so.0.0.3': No such file or directory
2020-02-07T09:09:05.3036366Z make[1]: *** [Makefile:30: all] Error 1
2020-02-07T09:09:05.3036674Z make[1]: Leaving directory '/kernel-collector/library'
2020-02-07T09:09:05.3038758Z make: *** [Makefile:24: ../library/libnetdata_ebpf.so.1.0] Error 2
2020-02-07T09:09:05.3043959Z ERROR: Build failed ...
2020-02-07T09:09:05.4745241Z ##[error]Process completed with exit code 1.
2020-02-07T09:09:05.4764658Z Cleaning up orphan processes
```

--- a/libbpf_0_0_1/src/libbpf.c
+++ b/libbpf_0_0_1/src/libbpf.c
@@ -22,6 +22,9 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <errno.h>
+// clang-format off
+#include <linux/stddef.h>
+// clang-format on
 #include <asm/unistd.h>
 #include <linux/err.h>
 #include <linux/kernel.h>

